query/physicalplan/aggregate.go

```diff
@@ -103,21 +107,26 @@ func chooseAggregationFunction(
 	}
 }
 
+// AggregateColumn groups together some lower level primitives to aggregate.
+type AggregateColumn struct {
+	expr       logicalplan.Expr
+	resultName string
+	function   AggregationFunction
+	arrays     []builder.ColumnBuilder // TODO: These can actually live outside this struct and be shared. Only at the very end will they be read by each column and then aggregated separately.
```

Comment:
Since `builder.ColumnBuilder` is an interface (i.e. pointer), I think these can actually stay here but each `AggregateColumn.arrays[i]` can point to the same underlying builder.

Reply:
I'm not 100% sure what you're referring to with "can actually stay here". Do you mean they can stay outside of the AggregateColumn in the `HashAggregate` struct?

Reply:
No, that `arrays` can stay as an `Aggregation` field, just that you can always set it up so that two aggregation struct array field indices point to the same builder.